src/storage/compactor/src/server.rs

```diff
@@ -414,12 +414,12 @@ pub async fn shared_compactor_serve(
                     tokio::select! {
                         _ = tokio::signal::ctrl_c() => {},
                         _ = &mut shutdown_recv => {
-                                if let Err(err) = shutdown_sender.send(()) {
-                                    tracing::warn!("Failed to send shutdown: {:?}", err);
-                                }
-                                if let Err(err) = join_handle.await {
-                                    tracing::warn!("Failed to join shutdown: {:?}", err);
-                                }
+                            if shutdown_sender.send(()).is_err() {
```

Comment:
ditto

Reply:
...if the payload is the unit type, then we can just ignore that.